doc/source/user/building.rst

```diff
@@ -64,75 +64,84 @@ Building NumPy requires the following software installed:
 
    Clone the repository following the instructions in :doc:`/dev/index`.
 
-Basic Installation
+.. note::
+
+    Starting on version 1.26, NumPy will adopt Meson as its build system (see
+    :ref:`distutils-status-migration` and
+    :doc:`scipy:building/understanding_meson` for more details.)
+
+Basic installation
 ------------------
 
-To install NumPy, run::
+To build and install NumPy from a local copy of the source code, run::
 
     pip install .
 
-To perform an in-place build that can be run from the source folder run::
+This will install all build dependencies and use Meson to compile and install
+the NumPy C-extensions and Python modules. If you need more control of build
+options and commands, see the following sections.
 
-    python setup.py build_ext --inplace
+Building from source
+~~~~~~~~~~~~~~~~~~~~
 
-*Note: for build instructions to do development work on NumPy itself, see*
-:ref:`development-environment`.
+To build NumPy with meson, you can use the
+`spin <https://github.com/scientific-python/spin>`_ utility. First, install the
+build requirements with
 
-Testing
--------
+::
 
-Make sure to test your builds. To ensure everything stays in shape, see if
-all tests pass.
+    python -m pip install -r build_requirements.txt
 
-The test suite requires additional dependencies, which can easily be 
-installed with::
+then run::
 
-    $ python -m pip install -r test_requirements.txt
+    spin build
 
-Run tests::
+This does not result in an in-place build, and to run the version of NumPy you
+have just built, you can use::
 
-    $ python runtests.py -v -m full
+    spin ipython  # alternatively, `spin python`
 
-For detailed info on testing, see :ref:`testing-builds`.
+and import NumPy as usual.
+
+For more information on the ``spin`` utility, use ``spin --help``.
 
 .. _parallel-builds:
 
 Parallel builds
 ~~~~~~~~~~~~~~~
 
-It's possible to do a parallel build with::
+It's possible to do a parallel build with the following command::
+
+    spin build -j 4
 
-    python setup.py build -j 4 install --prefix $HOME/.local
+This will compile NumPy on 4 CPUs and install it into ``./build-install``.
 
-This will compile numpy on 4 CPUs and install it into the specified prefix.
-to perform a parallel in-place build, run::
+Advanced building with Meson
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    python setup.py build_ext --inplace -j 4
+For more options including selecting compilers, setting custom compiler flags
+and controlling parallelism, see :doc:`scipy:building/compilers_and_options`
+(from the SciPy documentation.)
 
-The number of build jobs can also be specified via the environment variable
-``NPY_NUM_BUILD_JOBS``.
+Testing
+-------
 
-Choosing the fortran compiler
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Make sure to test your builds. To ensure everything stays in shape, see if
+all tests pass.
 
-Compilers are auto-detected; building with a particular compiler can be done
-with ``--fcompiler``.  E.g. to select gfortran::
```

Comment:
Perhaps it's worthwhile to note that environment variables can be set for this? `meson` honors `FC` and friends